Sync fork with an upstream

alexeykudinkin opened the pull request:
Seems like this fork has not been updated for about 2 years now. Pulling in changes from the upstream.

beorn7 committed:
Merge pull request #654 from prometheus/beorn7/push

Minor improvements to the push code

beorn7 committed:
Merge pull request #657 from prometheus/beorn7/registry

Make hash collisions in the registry much less likely

beorn7 committed:
Merge pull request #655 from prometheus/beorn7/test

Unflake TestGoCollectorGoroutines

beorn7 committed:
Fix function names in API examples

The part after the underscore has to be lowercase. Otherwise, it is
seen as the name of a Go type, which doesn't exist, which will result
in the example not showing up in godoc.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Update dependencies

Just to make sure we are up to date before releasing.
This only affects golang.org/x/sys.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Cut v1.2.0

One might argue that supporting the newest Pushgateway is just an
enhancement or even a bugfix (and this should be v1.1.1, which would
be a really nice version number). I decided to go for a positive view
of things and call it a feature. If something else changes its
behavior, it's not really a bug from client_golang's point of view.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #658 from prometheus/beorn7/release

Cut v1.2.0

beorn7 committed:
Use a cleaner initialization of `separatorByteSlice`

The `const separatorByte` wasn't used anymore actually. In `vec.go`,
we were using `model.SeparatorByte`, which is better anyway. So remove
the unused constant and initialize `separatorByteSlice` with
`model.SeparatorByte`, too.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #659 from prometheus/beorn7/nit

Use a cleaner initialization of `separatorByteSlice`

beorn7 committed:
Simplify code

`gofmt -s` taught me that this is possible.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #660 from prometheus/beorn7/nit

Simplify code

costela committed:
ensure same collectorID calculated on reg and unreg

Signed-off-by: Leo Antunes <leo@costela.net>

beorn7 committed:
Add more precise Go version requirements.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #663 from costela/master

Fix collectorID calculation for Unregister()

beorn7 committed:
Merge pull request #664 from prometheus/beorn7/doc

Add more precise Go version requirements.

beorn7 committed:
Cut v1.2.1

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #665 from prometheus/beorn7/release

Cut v1.2.1

jeffwidman committed:
Remove old reference to basicMetricVec

Contrary to the code comment, I see no `basicMetricVec` implementation.
Grep'ing this project shows this is the only reference. So I suspect
it's an outdated comment and can be removed to minimize confusion.

I'm unclear whether other parts of that comment are also incorrect and
need updating.

Signed-off-by: Jeff Widman <jeff@jeffwidman.com>

beorn7 committed:
Merge pull request #666 from jeffwidman/patch-1

Remove old reference to basicMetricVec

krasi-georgiev committed:
rmeove Krasi from the maintainers.

Signed-off-by: Krasi Georgiev <8903888+krasi-georgiev@users.noreply.github.com>

beorn7 committed:
Merge pull request #667 from krasi-georgiev/maintainers

remove Krasi from the maintainers.

kamaev committed:
add graphite tags

Signed-off-by: kamaev <timurkamaev@protonmail.com>

kamaev committed:
fix tests

Signed-off-by: kamaev <timurkamaev@protonmail.com>

beorn7 committed:
Merge pull request #668 from kamaev/master

Send Graphite metrics with tags

Dean-Coakley committed:
Fix link to writing exporters guide

Signed-off-by: Dean Coakley <dean.s.coakley@gmail.com>

beorn7 committed:
Merge pull request #686 from Dean-Coakley/patch-2

Fix link to writing exporters guide

beorn7 committed:
Fix typo

Signed-off-by: beorn7 <beorn@grafana.com>

cben committed:
api client examples: avoid shadowing package with variable

Signed-off-by: Beni Cherniavsky-Paskin <cben@redhat.com>

beorn7 committed:
Merge pull request #695 from cben/api-client-examples-shadowing

api client examples: avoid shadowing package with variable

beorn7 committed:
Improve doc comment for NewGaugeFunc

- Simplify confusing wording about concurrency safety.
- Add link to example for info metric.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #697 from prometheus/beorn7/doc

Improve doc comment for NewGaugeFunc

joe-elliott committed:
First pass unravel

Signed-off-by: Joe Elliott <number101010@gmail.com>

joe-elliott committed:
Refactor ~worked.  All tests passing except one

Signed-off-by: Joe Elliott <number101010@gmail.com>

joe-elliott committed:
Fixed TestDoGetFallback test

Signed-off-by: Joe Elliott <number101010@gmail.com>

joe-elliott committed:
Moved warnings into apiclient

Signed-off-by: Joe Elliott <number101010@gmail.com>

beorn7 committed:
Merge pull request #699 from joe-elliott/http-client-warnings

Return Prometheus Warnings

beorn7 committed:
Update dependencies

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Cut v1.3.0

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #700 from prometheus/beorn7/release

Update dependencies and cut v1.3.0

beorn7 committed:
Improve doc string for `go_gc_duration_seconds`

Fixes #618.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #702 from prometheus/beorn7/doc

Improve doc string for `go_gc_duration_seconds`

bwplotka committed:
Added testutil.CollectAndCount

Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

beorn7 committed:
Merge pull request #703 from prometheus/testutil-metric-count

Added `testutil.CollectAndCount`

beorn7 committed:
Add exemplars to counter and histogram

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Add tests for examplars

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #706 from prometheus/beorn7/exemplars

Add exemplars to counter and histogram

beorn7 committed:
Update dependencies

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Cut v1.4.0

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Pull out ...WithExemplar methods into separate interfaces

This is, sadly, the only way to avoid a breaking change. The cost is
that anyone using exemplars has to perform a type assertion. This is,
however, a common pattern where interfaces turn out to need additional
methods in a stable library or only some implementations can provide
the additional methods (AKA "interface upgrade").

Needless to say that in v2 of this library, we'll do things in a more
straight forward way.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #708 from prometheus/beorn7/exemplars

Pull out ...WithExemplar methods into separate interfaces

beorn7 committed:
Fix typo in doc comment

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #707 from prometheus/beorn7/release

Cut v1.4.0

shreyassrivatsan committed:
Initialize now fn for counters

Signed-off-by: Shreyas Srivatsan <shreyas@chronosphere.io>

beorn7 committed:
Merge pull request #710 from shreyassrivatsan/ss/fix

Initialize now fn for counter

beorn7 committed:
Cut v1.4.1

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #711 from prometheus/beorn7/release

Cut v1.4.1

beorn7 committed:
Add a "factory" to make promauto work for custom registries

Also, update the package documentation. The concerns about the global
registry aren't really valid anymore because promauto now also works
with custom registries.

The musings about the http.DefaultMux are more a digression and
shouldn't be in a doc comment.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Explicitly forward metricVec methods Collect, Describe, Reset

Interestingly, methods implicitly forwarded from embedded types are
detected by GoDoc if they are just one level deep. Embedded types in
the embedded type are not recognized. This commit therefore adds
explicit forwarding methods for Collect, Describe, and Reset.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Remove obsolete references to `Untyped` from doc comments

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #714 from prometheus/beorn7/doc

Explicitly forward metricVec methods Collect, Describe, Reset

beorn7 committed:
Add NewUntypedFunc to promauto

This constructor was simply forgotten.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #713 from prometheus/beorn7/promauto

Add a "factory" to make promauto work for custom registries

prombot committed:
makefile: update Makefile.common with newer version

Signed-off-by: prombot <prometheus-team@googlegroups.com>

simonpasquier committed:
Merge pull request #717 from prometheus/makefile_common

Synchronize Makefile.common from prometheus/prometheus

gotjosh committed:
API Client: Support new metadata endpoint in v1

Introduces support for the new metadata endpoint from Prometheus. The new endpoint provides information independent of targets and collapses the unique combinations of HELP, TYPE and UNIT.

Fixes #705

Signed-off-by: gotjosh <josue@grafana.com>

gotjosh committed:
Update api/prometheus/v1/api_test.go

Co-Authored-By: Tobias Guggenmos <tguggenm@redhat.com>
Signed-off-by: gotjosh <josue@grafana.com>

gotjosh committed:
Update api/prometheus/v1/api_test.go

Co-Authored-By: Tobias Guggenmos <tguggenm@redhat.com>
Signed-off-by: gotjosh <josue@grafana.com>

gotjosh committed:
Address review feedback

Signed-off-by: gotjosh <josue@grafana.com>

simonpasquier committed:
Add Circle CI configuration

Signed-off-by: Simon Pasquier <spasquie@redhat.com>

beorn7 committed:
Merge pull request #718 from gotjosh/support-metric-metadata-endpoint

API Client: Support new metadata endpoint in v1

beorn7 committed:
Cut v1.5.0

Signed-off-by: beorn7 <beorn@grafana.com>

bwplotka committed:
Merge pull request #720 from prometheus/beorn7/release

Cut v1.5.0

beorn7 committed:
Merge pull request #719 from simonpasquier/enable-circleci

Add Circle CI configuration

beorn7 committed:
Fix promhttp error handling

Essentially, just don't try to set a status code and send any error
message in the body once metrics gathering has succeeded. At that
point, the most likely reason for errors is anyway that the client has
disconnected, in which sending an error is moot. The other possible
reason for an error is a problem during metrics encoding. This is
unlikely to happen (it's a coding error here in client_golang in any
case), and if it is happening, the odds are we have already sent
something to the ResponseWriter, which means we cannot set a status
code anymore. The doc comment for HTTPErrorOnError now describes these
circumstances explicitly and recommends to set a logger to report that
kind of error.

This should fix the reason for the infamous `superfluous
response.WriteHeader call` message.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #726 from prometheus/beorn7/promhttp

Fix promhttp error handling

beorn7 committed:
Cut v1.5.1

Signed-off-by: beorn7 <beorn@grafana.com>

bwplotka committed:
Merge pull request #727 from prometheus/beorn7/release

Cut v1.5.1

Commit:
Improve API error handling

Signed-off-by: huanggze <loganhuang@yunify.com>

beorn7 committed:
Merge pull request #731 from huanggze/master

Improve API error handling

cobolbaby committed:
fix: process_resident_memory_bytes detected is incorrect in Win7 32bit

Ref: https://github.com/prometheus/client_golang/issues/728
Signed-off-by: Cobolbaby <Zhang.Xing-Long@inventec.com>

cobolbaby committed:
feat: Change processMemoryCounters struct uint declaration to uintptr

Signed-off-by: cobolbaby <cobolbaby@qq.com>

beorn7 committed:
Merge pull request #734 from cobolbaby/master

fix: process_resident_memory_bytes detected is incorrect in Win7 32bit

colega committed:
GaugeFunc: ConstLabels godoc example

Document the example usage of ConstLabels to register several GaugeFuncs
on the same metric name. Also reference it from the NewCounterFunc
documentation as it's similar.

Ref: https://github.com/prometheus/client_golang/pull/736

Signed-off-by: Oleg Zaytsev <mail@olegzaytsev.com>

RainbowMango committed:
Porting promlint from prometheus/prometheus.

Signed-off-by: RainbowMango <renhongcai@huawei.com>

RainbowMango committed:
fix static check warnings by Goland.

Signed-off-by: RainbowMango <renhongcai@huawei.com>

RainbowMango committed:
Fix unit test failing with Go1.14+

Signed-off-by: RainbowMango <renhongcai@huawei.com>

beorn7 committed:
Merge pull request #740 from RainbowMango/pr_fix_unit_test_failing_with_go1.14

Fix unit test failing with Go1.14+

beorn7 committed:
Add Go 1.14

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #741 from prometheus/beorn7/go-requirement

Add Go 1.14

beorn7 committed:
Merge pull request #738 from colega/colega/gauge-func-const-labels-godoc

GaugeFunc: ConstLabels godoc example

beorn7 committed:
Merge pull request #739 from RainbowMango/pr_porting_promlint

Porting promlint from prometheus/prometheus

simonpasquier committed:
Remove Travis CI

Signed-off-by: Simon Pasquier <spasquie@redhat.com>

beorn7 committed:
Add the capability to lint MetricFamilies directly

Also, change all the `dto.MetricFamily` arguments to pointers to be
more consistent with what we do in client_golang in general.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Add helper functions for linting to testutil

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #742 from simonpasquier/remove-travis

Remove Travis CI

beorn7 committed:
Improve various comments

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #743 from prometheus/beorn7/test

Add linter helpers to testutil

beorn7 committed:
Update dependencies

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Cut v1.6.0

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #744 from prometheus/beorn7/release

Update dependencies and cut v1.6.0

mcristina422 committed:
Fix typo in godoc help text

Signed-off-by: Michael Cristina <mcristina444@gmail.com>

beorn7 committed:
Merge pull request #745 from mcristina422/patch-1

Fix typo in godoc help text

beorn7 committed:
Properly handle empty job and label values

An empty job name was always an error, but it was previously only
detected when pushing to the PGW and receiving an error. With this
commit, the error is detected before pushing.

An empty label value should have been OK but was encoded in a way that
couldn't be pushed to the
PGW. Cf. https://github.com/prometheus/pushgateway/issues/344 . This
commit changes the creation of the path in the URL so that it works
with empty label values.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Improve CollectAndCount

Now that we have also added CollectAndLint and GatherAndLint, I
thought we should bring CollectAndCount in line. So:

- Add GatherAndCount.
- Add filtering by metric name.
- Add tests.

Minor wart: CollectAndCount should now return `(int, error)`, but that
would be a breaking change as the current version just returns
`int`. I decided to let the new version panic when it should return an
error. An error is anyway very unlikely, so the biggest annoyance here
is really just the inconsistency.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #752 from prometheus/beorn7/push

Properly handle empty job and label values

beorn7 committed:
Add lint:ignore for protobuf deprecation

`github.com/golang/protobuf/proto` is deprecated in lieu of
`google.golang.org/protobuf/proto`. However, we cannot simply
migrate. Types from the proto package are exposed to users of packages
in this repo. If we migrate here, users have to migrate to. Thus, we
could only migrate with a major version bump.

In different news, with all the inline lint:ignore comments, including
the existing ones, there is no need to repeat the exception in the
Makefile.

A current version of `staticcheck` is happy with the code after this
commit. golangci-lint is broken at the moment, however, and ignores
the lint:ignore comments in the code as well as those via envvar.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #753 from prometheus/beorn7/test

Improve CollectAndCount

beorn7 committed:
Merge pull request #754 from prometheus/beorn7/deprecation

Add lint:ignore for protobuf deprecation

lilic committed:
api/prometheus/v1/api.go: Add support for /runtimeinfo endpoint

Signed-off-by: Lili Cosic <cosiclili@gmail.com>

lilic committed:
api/prometheus/v1/api_test.go: Add test case for runtimeinfo endpoint

Signed-off-by: Lili Cosic <cosiclili@gmail.com>

prombot committed:
makefile: update Makefile.common with newer version

Signed-off-by: prombot <prometheus-team@googlegroups.com>

RainbowMango committed:
Bump github.com/golang/protobuf to v1.4.2

Signed-off-by: RainbowMango <renhongcai@huawei.com>

beorn7 committed:
Merge pull request #757 from RainbowMango/pr_bump_golang_protobuf_14_2

Bump github.com/golang/protobuf to v1.4.2

beorn7 committed:
Merge pull request #756 from prometheus/makefile_common

Synchronize Makefile.common from prometheus/prometheus

mdlayher committed:
testutil/promlint: allow Kelvin as a base unit for color temperature

Signed-off-by: Matt Layher <mdlayher@gmail.com>

beorn7 committed:
Merge pull request #761 from prometheus/mdl-kelvin

testutil/promlint: allow Kelvin as a base unit for color temperature

beorn7 committed:
Merge pull request #755 from lilic/add-runtimeinfo

api/prometheus/v1/api.go: Add support for status/runtimeinfo endpoint

beorn7 committed:
Improve doc comments in promauto

In particular, point out that `With(nil)` is valid.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #762 from prometheus/beorn7/doc

Improve doc comments in promauto

tomwilkie committed:
Ensure that nil registers are treat as a no-op, even when wrapping.

Signed-off-by: Tom Wilkie <tom.wilkie@gmail.com>

tomwilkie committed:
Review feedback: use one line.

Signed-off-by: Tom Wilkie <tom.wilkie@gmail.com>

tomwilkie committed:
Review feedback: add comment and tests for WrapRegistererWith.

Signed-off-by: Tom Wilkie <tom.wilkie@gmail.com>

beorn7 committed:
Merge pull request #764 from prometheus/wrap-nil

Ensure that nil registers are treat as a no-op, even when wrapping.

Nexucis committed:
add start/end parameter for LabelNames

Signed-off-by: Augustin Husson <husson.augustin@gmail.com>

Nexucis committed:
add start/end parameter for LabelValues

Signed-off-by: Augustin Husson <husson.augustin@gmail.com>

beorn7 committed:
Merge pull request #767 from Nexucis/feature/labelNames-parameter

add start/end parameter for LabelNames

beorn7 committed:
Update dependencies

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Cut v1.7.0

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #769 from prometheus/beorn7/release

Cut v1.7.0

yeya24 committed:
fix time parameter propagation in labels API

Signed-off-by: yeya24 <yb532204897@gmail.com>

mhansen committed:
histogram.go: fix copy/paste typo

This method calls NewConstHistogram.

Signed-off-by: Mark Hansen <mark@markhansen.co.nz>

beorn7 committed:
Merge pull request #772 from mhansen/patch-1

histogram.go: fix copy/paste typo

beorn7 committed:
Merge pull request #771 from yeya24/fix-time-range-propagation

Fix time parameter propagation in labels API

prombot committed:
Update common Prometheus files

Signed-off-by: prombot <prometheus-team@googlegroups.com>

beorn7 committed:
Merge pull request #774 from prometheus/repo_sync

Synchronize common files from prometheus/prometheus

beorn7 committed:
Cut v1.7.1

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #775 from prometheus/beorn7/release

Cut v1.7.1

HimaVarsha94 committed:
Add support for tsdb endpoint (#773)

Signed-off-by: Hima Varsha <hdureddy@apple.com>

mxey committed:
Use time.Time for timestamps in Runtimeinfo (#777)

Signed-off-by: Maximilian Gaß <m.gass@babiel.com>

johejo committed:
fix tests warning about string(int) type conversions

See https://github.com/golang/go/issues/32479

Signed-off-by: Mitsuo Heijo <mitsuo.heijo@gmail.com>

beorn7 committed:
Merge pull request #779 from johejo/fix_tests_string_type_conversions

fix tests warning about string(int) type conversions

Commit:
Update collector comment about GC stop-the-world

Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

Commit:
remove reference to metrics after reslicing

Signed-off-by: Dima Kozlov <hummerd@mail.ru>

beorn7 committed:
Merge pull request #783 from roidelapluie/gc15

Update collector comment about GC stop-the-world

beorn7 committed:
Merge pull request #784 from hummerd/master

Remove reference to Metric after reslicing

Commit:
Support go 1.15

Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

beorn7 committed:
Merge pull request #792 from roidelapluie/go115

Support go 1.15

johncming committed:
Replace with the standard library constant.

Signed-off-by: johncming <johncming@yahoo.com>

beorn7 committed:
Merge pull request #793 from johncming/stdlib-constant

Replace with the standard library constant.

prombot committed:
Update common Prometheus files

Signed-off-by: prombot <prometheus-team@googlegroups.com>

beorn7 committed:
Merge pull request #797 from prometheus/repo_sync

Synchronize common files from prometheus/prometheus

beorn7 committed:
Remove spurious commas from links to the docs site

I assume older Nanoc versions rendered the anchors with commas, but
the current doesn't.

Also, this adds the same link to another doc comment where it is also
relevant.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #800 from prometheus/beorn7/doc

Remove spurious commas from links to the docs site

beorn7 committed:
API client: Enable fallback on status code 501, too

When discussing #801, I remembered #794. While dealing with the
latter, I read the HTTP RFC, stumbling upon the following:

    When a request method is received
    that is unrecognized or not implemented by an origin server, the
    origin server SHOULD respond with the 501 (Not Implemented) status
    code.  When a request method is received that is known by an origin
    server but not allowed for the target resource, the origin server
    SHOULD respond with the 405 (Method Not Allowed) status code.

Concluding from that, it is possible that a server desiring a fallback
to GET will send a status code of 501. It is even preferred if that
server does not offer any resource to be used with the POST method.

Therefore, I think we should fallback to GET on a 501, too.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #802 from prometheus/beorn7/fallback

API client: Enable fallback on status code 501, too

beorn7 committed:
Export MetricVec (again)

MetricVec was already exported in early versions of this library, but
nobody really used it to implement vectors of custom Metric
implementations. Now #796 has shown up with a fairly special use case
for which I'd prefer a custom implementation of a special
"auto-sampling histogram" outside of this library. Therefore, I'd like
to reinstate support for creating vectors of custom Metric
implementations.

I played around for quite some while with the option of a separate
package providing the tools one would need to create vectors of custom
Metric implementations. However, with the current structure of the
prometheus/client_golang/prometheus package, this leads to a lot of
complications with circular dependencies. (The new package would need
the primitives from the prometheus package, while the existing metric
vectors like GaugeVec need to import the new vector package to not
duplicate the implementation. Separating vector types from the main
prometheus package is out of the question at this point because that
would be a breaking change.)

Signed-off-by: beorn7 <beorn@grafana.com>

sbookworm committed:
add the NewPidFileFn to helper

Signed-off-by: sbookworm <lisong.cruise@gmail.com>

beorn7 committed:
Merge pull request #803 from prometheus/beorn7/vec

Export MetricVec (again)

beorn7 committed:
Update dependencies

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Cut v1.8.0

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Run check for unused/missing Go packages only against latest Go version

The tidied up `go.mod` file tends to change between Go versions.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #806 from prometheus/beorn7/release

Cut v1.8.0

prombot committed:
Update common Prometheus files

Signed-off-by: prombot <prometheus-team@googlegroups.com>

beorn7 committed:
Merge pull request #804 from sbookworm/get_pid_from_file

add the NewPidFileFn to helper

beorn7 committed:
Merge pull request #809 from prometheus/repo_sync

Synchronize common files from prometheus/prometheus

prombot committed:
Update common Prometheus files

Signed-off-by: prombot <prometheus-team@googlegroups.com>

beorn7 committed:
Merge pull request #811 from prometheus/repo_sync

Synchronize common files from prometheus/prometheus

bwplotka committed:
Added example api code showing how to add auth tokens and user agents to prom client. (#817)

* Added example api code showing how to add auth tokens and user agents to prom client.

Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

* Ran go mod tidy.

Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

jubalh committed:
Correct spelling: possibilites -> possibilities

Signed-off-by: Michael Vetter <jubalh@iodoru.org>

beorn7 committed:
Merge pull request #819 from jubalh/sp

Correct spelling: possibilites -> possibilities

beorn7 committed:
Be more explicit about the multi-line properties of MultiError

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #821 from prometheus/beorn7/multierror

Be more explicit about the multi-line properties of MultiError

beorn7 committed:
promhttp: Correctly detect invalid metric and label names

Without this fix, the `InstrumentHandler...` middlewares get locked in
an endless loop in case of an invalid Collector, eating all the memory.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #823 from prometheus/beorn7/promhttp

promhttp: Correctly detect invalid metric and label names

beorn7 committed:
Update dependencies

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Cut release 1.9.0

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #826 from prometheus/beorn7/release

￼ Cut release 1.9.0

yeya24 committed:
support matchers in labels API

Signed-off-by: yeya24 <yb532204897@gmail.com>

Commit:
Add SECURITY.md

This commit adds a security policy to this repository. SECURITY.md files
are threated in a special way by GitHub, helping users to know how to
best submit security issues for the projects.

In this case, we simply point to our existing documentation on
prometheus.io.

The content of this file will be synced automatically with the
prometheus/prometheus repository, as our security policy covers all the
repositories. This sync is automated with prombot, like other files
(LICENSE, Makefile.common).

https://docs.github.com/en/free-pro-team@latest/github/managing-security-vulnerabilities/adding-a-security-policy-to-your-repository
Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

beorn7 committed:
Merge pull request #831 from roidelapluie/security-dot-md

Add SECURITY.md

zhsj committed:
Bump prometheus/procfs to 0.3.0 to fix building on riscv64

This picks up https://github.com/prometheus/procfs/pull/325

Signed-off-by: Shengjing Zhu <i@zhsj.me>

beorn7 committed:
Merge pull request #833 from zhsj/riscv64

Bump prometheus/procfs to 0.3.0 to fix building on riscv64

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/m3db/prometheus_client_golang?pullRequest=1) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/m3db/prometheus_client_golang?pullRequest=1) before we can accept your contribution.<br/>**0** out of **17** committers have signed the CLA.<br/><br/>:x: prombot<br/>:x: beorn7<br/>:x: yeya24<br/>:x: Dean-Coakley<br/>:x: gouthamve<br/>:x: johejo<br/>:x: SuperQ<br/>:x: Ben Ye<br/>:x: gozeloglu<br/>:x: sivabalan<br/>:x: sbunce<br/>:x: dtrudg<br/>:x: kakkoyun<br/>:x: mrueg<br/>:x: prymitive<br/>:x: mknyszek<br/>:x: jcchavezs<br/><hr/>**Ben Ye** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/m3db/prometheus_client_golang?pullRequest=1) it.</sub>

yeya24 committed:
update comment

Signed-off-by: yeya24 <yb532204897@gmail.com>

Commit:
Fix typo in comments

Signed-off-by: debiandebiandebian <debiandebiandebiandebian@gmail.com>

beorn7 committed:
Merge pull request #835 from debiandebiandebian/fix-typo-in-comments

Fix typo in comments

beorn7 committed:
Merge pull request #828 from yeya24/support-label-matchers

Support matchers in labels API

ntk148v committed:
Add buildinfo method

Co-authored-by: Lili Cosic <cosiclili@gmail.com>
Signed-off-by: Kien Nguyen <kiennt2609@gmail.com>

beorn7 committed:
Update dependencies

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Bump minumim required Go version to 1.13

Since 1.16 is out, we still support the last four minor releases.

The bump was required by the prometheus/procfs package using the new
`%w` printf directives. However, it also allows us to remove some
special casing about build info.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #841 from ntk148v/buildinfo-api

Add buildinfo method

beorn7 committed:
Merge pull request #843 from prometheus/beorn7/release

Update dependencies

prombot committed:
Update common Prometheus files

Signed-off-by: prombot <prometheus-team@googlegroups.com>

beorn7 committed:
Fix linter ignores

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #844 from prometheus/repo_sync

Synchronize common files from prometheus/prometheus

beorn7 committed:
Cut v1.10.0

Signed-off-by: beorn7 <beorn@grafana.com>

prombot committed:
Update common Prometheus files

Signed-off-by: prombot <prometheus-team@googlegroups.com>

beorn7 committed:
Merge pull request #845 from prometheus/beorn7/release

Cut v1.10.0

beorn7 committed:
Merge pull request #846 from prometheus/repo_sync

Synchronize common files from prometheus/prometheus

prombot committed:
Update common Prometheus files

Signed-off-by: prombot <prometheus-team@googlegroups.com>

beorn7 committed:
Merge pull request #849 from prometheus/repo_sync

Synchronize common files from prometheus/prometheus

prombot committed:
Update common Prometheus files

Signed-off-by: prombot <prometheus-team@googlegroups.com>

beorn7 committed:
Merge pull request #853 from prometheus/repo_sync

Synchronize common files from prometheus/prometheus

gouthamve committed:
Add newer fields to Rules API

Signed-off-by: Goutham Veeramachaneni <gouthamve@gmail.com>

yeya24 committed:
add missing fields to targets API

Signed-off-by: yeya24 <yb532204897@gmail.com>

beorn7 committed:
Merge pull request #855 from gouthamve/add-new-elements-to-rules

Add newer fields to Rules API

beorn7 committed:
Merge pull request #856 from yeya24/targets-field

Add missing fields to targets API

prombot committed:
Update common Prometheus files

Signed-off-by: prombot <prometheus-team@googlegroups.com>

beorn7 committed:
Merge pull request #857 from prometheus/repo_sync

Synchronize common files from prometheus/prometheus

yeya24 committed:
add exemplars API support

Signed-off-by: yeya24 <yb532204897@gmail.com>

johejo committed:
Add new collectors package

Signed-off-by: Mitsuo Heijo <mitsuo.heijo@gmail.com>

beorn7 committed:
Merge pull request #861 from yeya24/add-exemplars

Add exemplars API support

Dean-Coakley committed:
Improve description of MaxAge in summary docs (#864)

* Improve description of MaxAge in summary docs

Signed-off-by: Dean Coakley <dean.s.coakley@gmail.com>

beorn7 committed:
Merge pull request #862 from johejo/new_collector_package

Add new collectors package

johejo committed:
Add collector for database/sql#DBStats (#866)

* Add collector for database/sql#DBStats

Signed-off-by: Mitsuo Heijo <mitsuo.heijo@gmail.com>

beorn7 committed:
Make dbStatsCollector more DRY

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #867 from prometheus/beorn7/collectors

Make dbStatsCollector more DRY

beorn7 committed:
Document implications of negative observations

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Change maintainers from @beorn7 to @bwplotka/@kakkoyun (#873)

Lazy consensus was already established among prometheus-team. This
change will be announced on prometheus-developers@ once it is merged.

Signed-off-by: beorn7 <beorn@grafana.com>

beorn7 committed:
Merge pull request #871 from prometheus/beorn7/doc

Document implications of negative observations

SuperQ committed:
Update Go modules

Update all modules to the latest versions. This starts to reduce the
indirect dependencies on go-kit/kit in favor of go-kit/log.

Signed-off-by: SuperQ <superq@gmail.com>

SuperQ committed:
Merge pull request #875 from prometheus/superq/update_mods

Update Go modules

kakkoyun committed:
Cut v1.11.0 (#877)

Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

kakkoyun committed:
Address minor issues on Changelog

Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

kakkoyun committed:
Merge pull request #879 from kakkoyun/fix_changelog

Address minor issues on the changelog

sivabalan committed:
Updating dependency versions

Signed-off-by: Sivabalan Thirunavukkarasu <s.thirunavukkarasu@salesforce.com>

SuperQ committed:
Update status badgets

Update to new locations
* Switch to CircleCI build badge.
* Use new pkg.go.dev for docs.

Signed-off-by: SuperQ <superq@gmail.com>

prombot committed:
Update common Prometheus files

Signed-off-by: prombot <prometheus-team@googlegroups.com>

kakkoyun committed:
Merge pull request #888 from prometheus/repo_sync

Synchronize common files from prometheus/prometheus

kakkoyun committed:
Merge pull request #885 from prometheus/superq/badgers

Update status badgets

kakkoyun committed:
Merge pull request #881 from sivabalan/master

Updating dependency versions + cleanup

prombot committed:
Update common Prometheus files (#892)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

Commit:
add ExponentialBucketsRange function (#899)

This function calculates exponential buckets with different arguments
than the existing ExponentialBuckets function. Instead of specifying the
start and factor, the user can specify the min and max bucket value. We
have been doing it this way internally at my company for some time.

Signed-off-by: Seth Bunce <seth.bunce@getcruise.com>

prombot committed:
Update common Prometheus files

Signed-off-by: prombot <prometheus-team@googlegroups.com>

jcchavezs committed:
chore: adds support for go 1.17 on CI.

Signed-off-by: José Carlos Chávez <jcchavezs@gmail.com>

kakkoyun committed:
Merge pull request #909 from prometheus/repo_sync

Synchronize common files from prometheus/prometheus

dtrudg committed:
Update cespare/xxhash dependency (#913)

Brings in a fix for asm errors when dynamic linking with Go 1.17.

Fixes #912

Signed-off-by: David Trudgian <david.trudgian@sylabs.io>

prombot committed:
Update common Prometheus files

Signed-off-by: prombot <prometheus-team@googlegroups.com>

beorn7 committed:
Move flags and metrics into main()

Fixes #933.

Signed-off-by: beorn7 <beorn@grafana.com>

kakkoyun committed:
Merge pull request #935 from prometheus/beorn7/examples

example/random: Move flags and metrics into main()

gozeloglu committed:
Fix typo

Signed-off-by: Gökhan Özeloğlu <gozeloglu@gmail.com>

beorn7 committed:
Merge pull request #939 from gozeloglu/fix-typo

Fix typo

Commit:
support wal replay status api

Signed-off-by: Ben Ye <ben.ye@bytedance.com>

mrueg committed:
Add new go:build lines for go 1.17

Signed-off-by: Manuel Rüger <manuel@rueg.eu>

kakkoyun committed:
Merge pull request #950 from mrueg/go-1.17

Add support for go 1.17

kakkoyun committed:
Merge pull request #928 from prometheus/repo_sync

Synchronize common files from prometheus/prometheus

prombot committed:
Update common Prometheus files

Signed-off-by: prombot <prometheus-team@googlegroups.com>

kakkoyun committed:
Merge pull request #952 from prometheus/repo_sync

Synchronize common files from prometheus/prometheus

kakkoyun committed:
Merge pull request #944 from yeya24/add-wal-replay-status

API: support wal replay status api

prymitive committed:
Update /api/v1/status/tsdb to include headStats (#925)

Current structs for v1 HTTP API assume that there is chunkCount and timeSeriesCount on /api/v1/status/runtimeinfo, which seems to be gone for at least a few releases. /api/v1/status/tsdb now holds an extra headStats block with chunkCount, numSeries and a few other fields.
Update API models and tests to reflect this change.

Signed-off-by: Łukasz Mierzwa <l.mierzwa@gmail.com>

mknyszek committed:
Use the runtime/metrics package for the Go collector for 1.17+ (#955)

This change introduces use of the runtime/metrics package in place of
runtime.MemStats for Go 1.17 or later. The runtime/metrics package was
introduced in Go 1.16, but not all the old metrics were accounted for
until 1.17.

The runtime/metrics package offers several advantages over using
runtime.MemStats:
* The list of metrics and their descriptions are machine-readable,
  allowing new metrics to get added without any additional work.
* Detailed histogram-based metrics are now available, offering much
  deeper insights into the Go runtime.
* The runtime/metrics API is significantly more efficient than
  runtime.MemStats, even with the additional metrics added, because
  it does not require any stop-the-world events.

That being said, integrating the package comes with some caveats, some
of which were discussed in #842. Namely:
* The old MemStats-based metrics need to continue working, so they're
  exported under their old names backed by equivalent runtime/metrics
  metrics.
* Earlier versions of Go need to continue working, so the old code
  remains, but behind a build tag.

Finally, a few notes about the implementation:
* This change includes a whole bunch of refactoring to avoid significant
  code duplication.
* This change adds a new histogram metric type specifically optimized
  for runtime/metrics histograms. This type's methods also include
  additional logic to deal with differences in bounds conventions.
* This change makes a whole bunch of decisions about how runtime/metrics
  names are translated.
* This change adds a `go generate` script to generate a list of expected
  runtime/metrics names for a given Go version for auditing. Users of
  new versions of Go will transparently be allowed to use new metrics,
  however.

Signed-off-by: Michael Anthony Knyszek <mknyszek@google.com>
